apps/explorer/lib/explorer/smart_contract/reader.ex

```diff
@@ -315,43 +315,18 @@ defmodule Explorer.SmartContract.Reader do
     end
   end
 
-  defp get_abi_with_method_id(abi) do
-    parsed_abi =
-      abi
-      |> ABI.parse_specification()
-
-    abi_with_method_id =
-      abi
-      |> Enum.map(fn target_method ->
-        methods =
-          parsed_abi
-          |> Enum.filter(fn method ->
-            Atom.to_string(method.type) == Map.get(target_method, "type") &&
-              method.function == Map.get(target_method, "name") &&
-              Enum.count(method.input_names) == Enum.count(Map.get(target_method, "inputs")) &&
-              input_types_matched?(method.types, target_method)
-          end)
-
-        if Enum.count(methods) > 0 do
-          method = Enum.at(methods, 0)
-          method_id = Map.get(method, :method_id)
-          method_with_id = Map.put(target_method, "method_id", Base.encode16(method_id, case: :lower))
-          method_with_id
-        else
-          target_method
-        end
-      end)
-
-    abi_with_method_id
-  end
-
-  defp input_types_matched?(types, target_method) do
-    types
-    |> Enum.with_index()
-    |> Enum.all?(fn {target_type, index} ->
-      type_to_compare = Map.get(Enum.at(Map.get(target_method, "inputs"), index), "type")
-      target_type_formatted = format_type(target_type)
-      target_type_formatted == type_to_compare
+  def get_abi_with_method_id(abi) do
+    abi
+    |> Enum.map(fn method ->
+      with parsed_method <- [method] |> ABI.parse_specification() |> Enum.at(0),
+           true <- is_map(parsed_method),
```

Comment:
You should be able to use guard clauses to prevent having to match on `true = is_map`

like this

```
    |> Enum.map(fn method ->
      with [parsed_method] when is_map(parsed_method) <- [method] |> ABI.parse_specification(),
           method_id when not is_nil(method_id) <- Map.get(parsed_method, :method_id) do
```